source/connection.c

```diff
@@ -439,38 +493,55 @@ struct aws_http_server *aws_http_server_new(const struct aws_http_server_options
     return server;
 
 error:
-    aws_http_server_destroy(server);
+    aws_http_server_release(server);
```

Comment:
Call `s_http_server_clean_up(server);` directly. It's boils to the same thing as server_release(), but it's more obvious that you're just cleaning up memory here.

Reply:
I change it to clean up in different states, because the mutex lock is not safe to clean up, if the init failed, and I can find a way to check if the mutex lock init fail or not in the s_http_server_clean_up(server); function. So do the hash table.